commit 7532c7d0138a4984eb0c8b12b9c310cf8a6d6977
Author: Amitkumar Karwar <akarwar@marvell.com>
Date:   Thu Jan 17 17:41:57 2013 -0800

    mwifiex: correction in status codes used for association failure
    
    When AP responds with appropriate status code, we forward that
    code correctly to cfg80211. But sometimes when there is no
    response from AP, our firmware uses proprietary status codes.
    We will map authentication timeout to WLAN_STATUS_AUTH_TIMEOUT
    and other proprietary codes to WLAN_STATUS_UNSPECIFIED_FAILURE.
    
    Signed-off-by: Amitkumar Karwar <akarwar@marvell.com>
    Signed-off-by: Bing Zhao <bzhao@marvell.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/mwifiex/fw.h b/drivers/net/wireless/mwifiex/fw.h
index 41c85dd78084..ebe2f6a7984c 100644
--- a/drivers/net/wireless/mwifiex/fw.h
+++ b/drivers/net/wireless/mwifiex/fw.h
@@ -330,6 +330,9 @@ enum P2P_MODES {
 #define HOST_SLEEP_CFG_GPIO_DEF		0xff
 #define HOST_SLEEP_CFG_GAP_DEF		0
 
+#define MWIFIEX_TIMEOUT_FOR_AP_RESP		0xfffc
+#define MWIFIEX_STATUS_CODE_AUTH_TIMEOUT	2
+
 #define CMD_F_HOSTCMD           (1 << 0)
 #define CMD_F_CANCELED          (1 << 1)